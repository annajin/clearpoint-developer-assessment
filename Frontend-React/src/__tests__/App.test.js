/* eslint-disable testing-library/no-unnecessary-act */
import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import renderer, { act } from 'react-test-renderer'
import App from '../App'

describe('App tests', () => {
  const renderElement = async() => {
    await act(async () => render(<App />))
  }

  test('renders static text', async () => {
    await renderElement();

    const footerElement = screen.getByText(/clearpoint.digital/i)
    expect(footerElement).toBeInTheDocument()
    const requirementText = screen.getByText("Add the ability to add (POST) a Todo Item by calling the backend API")
    expect(requirementText).toBeInTheDocument()
    const addItemTitle = screen.getAllByText("Add Item")
    expect(addItemTitle.length).toBeGreaterThan(0)
  })

  test('should match snapshot', () => {
    const component = renderer.create(<App />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
