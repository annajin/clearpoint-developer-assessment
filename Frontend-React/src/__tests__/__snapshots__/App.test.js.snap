// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <img
          className="img-fluid rounded"
          src="clearPointLogo.png"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="fade alert alert-success show"
          role="alert"
        >
          <div
            className="alert-heading h4"
          >
            Todo List App
          </div>
          Welcome to the ClearPoint frontend technical test. We like to keep things simple, yet clean so your task(s) are as follows:
          <br />
          <br />
          <ol
            className="list-left"
          >
            <li>
              Add the ability to add (POST) a Todo Item by calling the backend API
            </li>
            <li>
              Display (GET) all the current Todo Items in the below grid and display them in any order you wish
            </li>
            <li>
              Bonus points for completing the 'Mark as completed' button code for allowing users to update and mark a specific Todo Item as completed and for displaying any relevant validation errors/ messages from the API in the UI
            </li>
            <li>
              Feel free to add unit tests and refactor the component(s) as best you see fit
            </li>
          </ol>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="container"
        >
          <h1>
            Add Item
          </h1>
          <div
            className="mb-3 row"
          >
            <label
              className="form-label col-form-label col-sm-2"
              htmlFor="formAddTodoItem"
            >
              Description
            </label>
            <div
              className="col-md-6"
            >
              <input
                className="form-control"
                data-testid="text-input"
                id="formAddTodoItem"
                onChange={[Function]}
                placeholder="Enter description..."
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="mb-3 offset-md-2 row"
          >
            <div
              className="hstack gap-2 minBreakpoint-xs"
            >
              <button
                className="btn btn-primary"
                data-testid="add-button"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Add Item
              </button>
              <button
                className="btn btn-secondary"
                data-testid="clear-button"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Clear
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h1>
          Showing 
          0
           Item(s)
           
          <button
            className="pull-right btn btn-primary"
            data-testid="refresh-button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Refresh
          </button>
        </h1>
        <table
          className="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                Id
              </th>
              <th>
                Description
              </th>
              <th>
                Action
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
  <footer
    className="page-footer font-small teal pt-4"
  >
    <div
      className="footer-copyright text-center py-3"
    >
      Â© 2021 Copyright:
      <a
        href="https://clearpoint.digital"
        rel="noreferrer"
        target="_blank"
      >
        clearpoint.digital
      </a>
    </div>
  </footer>
</div>
`;
